### PR TITLE
Add contents for #51

### PR DESCRIPTION
**Pull request by hailiang-wang**
# Add post
related with 
https://github.com/flossway/flossway/issues/51

### CODE REVIEW

**Comment by hailiang-wang** [comment]
@tisonkun, please help review, thanks.

File: source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
+title: 开源加云原生，企业软件的大规模进化之路
```
Comment (tisonkun) [inline source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:2]:
```suggestion
title: 开源 + 云原生：企业软件的大规模进化之路
```

**Review by tisonkun — COMMENTED** [review]
I'd suggest an alternative title.

There are several format issues I'd like to address but it would be better I submit a patch based on this PR later and ask you for review which is more efficient.

File: source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md (hunk 1 above)
Comment (hailiang-wang) [reply source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:2]: Sure, a better choice for the title, I like it too.

**Comment by hailiang-wang** [comment]
image

The image links are broken, but it works fine with `npm run server`, the links would be valid with `https://flossway.github.io/flossway/` path as the preview, `https://flossway-s85a0ptsg-flossway.vercel.app/2022/03/23/oss-plus-cloudnative-large-dist-enterprise-software/` has omitted `flossway/` in path, which is the reason.

**Comment by tisonkun** [comment]
@hailiang-wang you can refer to https://hexo.io/docs/asset-folders for docs and https://github.com/flossway/flossway/blob/main/source/_posts/the-cathedral-recap.md for an example.

**Comment by hailiang-wang** [comment]
Now it looks good in preview mode.

**Comment by hailiang-wang** [comment]
About the title, "开源加云原生" OR "开源 + 云原生", I prefer the former after more thinking.
In the article, I use  "开源加云原生"  as a concept, it is a single term in the context, which can be indexed by search engine. On the other hand,   "开源 + 云原生" which would not have that effect, because "+" is omitted as a punctuation.

@tisonkun

File: source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+# 为什么是开源加云原生模式
+
+作为任何企业应用级别的软件，比如 CRM、ERP、智能客服系统，要以一种模式帮助上千万的企业上线，实际上是满足了一种新的供需平衡 - ==比以前的模式容纳了 10 倍以上的供需双方==。开源加云原生满足了这种供需关系，这个结论的论证包含以下三个方面：
+
+{% asset_img screenshot-20220323-190603.jpg "The philosophy underneath" %}
+
+<!-- * 开源加云原生实现精细化分工 -->
+<!-- * 开源加云原生带来飞轮效应 -->
+<!-- * 开源加云原生促进需求释放 -->
```
Comment (tisonkun) [inline source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:56]:
```suggestion
```

I think we don't have to keep these comments

File: source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+进入二十一世纪，随着移动互联网的普及，让更多的企业依赖软件应用发展业务，从客服、客户管理、仓储物流、财务等，都需要数字化，并且在不同季节还会出现淡季和旺季，对于大量的中小型企业，就需要租用这些企业软件，相当于是中小型企业团购软件服务，按量或按时间付费，此种模式就是 SaaS。这种方法降低了企业购置软件的门槛，每个行业的数字化企业进一步增多，从数十万，增长到数百万。这时，软件交付又变成了产品供应商。==这是我们看到的企业服务软件中的第二次大分工。==
+
+在中小企业发展过程中，因为其所处的竞争环境，其对创新和稳定的诉求，依然有强烈的定制化的渴望，SaaS 模式下，企业之间大量同质化，如果使用 SaaS 后进行定制，其成本相当于采购私有部署并进行定制的方案，方案不可行。所以，必然出现一种新的方案，时间正是现在。==在进入 2020s，第三次大分工开始了==，这次的目标是更大量的企业开始采购可迭代更新的、定制化友好的软件方案。这次，又回到了产品供应商、集成商一起为企业交付的模式下：1）产品供应商成为云服务平台；2）基础软件产品开源，降低 IT 集成商准入门槛，独立开发者也可以通过开源软件，为企业上线定制的软件服务。回到此种格局下，可以让不同的角色都参与企业软件交付中。类似的精细化的分工，已经在不同行业出现，比如滴滴之于传统的出租车公司，滴滴更加侧重于实现调度平台，而不是雇佣全职的司机和保养汽车。
+
+以上不同的代表企业分别是 SAP、Neusoft、Salesforce 和 Odoo。
+
+{% asset_img screenshot-20220323-110840.jpg "Delivery Trend for Enterprise Softwares" %}
+
+==只有使用开源加云原生的模式，才能召集足够规模的集成商和产品提供商一起，满足更大数量级的企业客户需求。==
+
```
Comment (tisonkun) [inline source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:73]:
```suggestion
```

At most one blank line. Ditto others. Please give a through to check.

File: source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+## 飞轮效应
+
+飞轮效应，就是可以持续不断的为企业客户提供好的软件，满足企业对物美价廉的诉求，满足企业对软件迭代更新的诉求。
+
+飞轮效应可以想象为，产品提供商和集成商服务于企业软件市场像转动一个巨大的轮子，开始转动会慢，但是随着时间的积累，轮子越转越快，形成长期稳定经营的效果。企业客户不但获得更好的体验，而且使用产品的生命周期比已有的其他模式长。
+
+那么，开源加云原生，是怎么做到这样的呢？
+
+首先，==最开始开源软件的价格低==，开源软件价格低的原因是软件的开发过程，由开源社区承担，产品供应商在领导软件研发，相比于闭源企业，开源社区在渠道、反馈机制、研发费用上，具有成本优势。比如传统闭源厂商的市场、销售和商务流程上的时间和金钱，在开源社区是可以节约的。那么价格低，就可以让开源加云原生获得第一批企业用户。
```
Comment (tisonkun) [inline source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:83]:
```suggestion
首先，**最开始开源软件的价格低**，开源软件价格低的原因是软件的开发过程，由开源社区承担，产品供应商在领导软件研发，相比于闭源企业，开源社区在渠道、反馈机制、研发费用上，具有成本优势。比如传统闭源厂商的市场、销售和商务流程上的时间和金钱，在开源社区是可以节约的。那么价格低，就可以让开源加云原生获得第一批企业用户。

```

I'd prefer bold than background highlight. But it's up to you.

Comment (hailiang-wang) [reply source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:78]: Will refine the highlights with both approaches.

**Comment by hailiang-wang** [comment]
About the highlights, revise as this -
* for sentences, background highlight
* for single terms, bold

File: source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+5. [精细化分工理论，科学管理之父泰勒](https://baike.sogou.com/v3139958.htm)
+
+6. [15年覆盖100万+企业，腾讯企点全通路融合，拿下赛道Top1，2021](https://www.sohu.com/a/491351696_202972)
+
+7. [The Odoo Story, 2014](https://www.odoo.com/blog/odoo-news-5/the-odoo-story-56)
+
+8. [Odoo, the Open Source ERP Leader, Announces €180M New Investment from Summit Partners, 2021](https://www.summitpartners.com/news/odoo-the-open-source-erp-leader-announces-180m-new-investment-from-summit-partners)
+
+9. [Odoo, an open-source business management software developer, at a $2.3B+ valuation, 2021](https://techcrunch.com/2021/07/29/summit-invests-215m-into-odoo-an-open-source-business-management-software-developer-at-a-2-3b-valuation/ )
```
Comment (tisonkun) [inline source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:132]:
```suggestion
1. [2021 年中国小企业数字初始化指数 2.0 白皮书](https://i.dell.com/sites/csdocuments/Shared-Content_data-Sheets_Documents/zh/cn/delldigitalizationbasic.pdf)
2. [红杉中国：2021企业数字化年度指南](https://byteclicks.com/28909.html)
3. [The Flywheel effect by Jim Collins](https://www.jimcollins.com/concepts/the-flywheel.html)
4. [时间机器理论, 软银孙正义](https://zhuanlan.zhihu.com/p/88770582)
5. [精细化分工理论，科学管理之父泰勒](https://baike.sogou.com/v3139958.htm)
6. [15年覆盖100万+企业，腾讯企点全通路融合，拿下赛道Top1，2021](https://www.sohu.com/a/491351696_202972)
7. [The Odoo Story, 2014](https://www.odoo.com/blog/odoo-news-5/the-odoo-story-56)
8. [Odoo, the Open Source ERP Leader, Announces €180M New Investment from Summit Partners, 2021](https://www.summitpartners.com/news/odoo-the-open-source-erp-leader-announces-180m-new-investment-from-summit-partners)
9. [Odoo, an open-source business management software developer, at a $2.3B+ valuation, 2021](https://techcrunch.com/2021/07/29/summit-invests-215m-into-odoo-an-open-source-business-management-software-developer-at-a-2-3b-valuation/ )
```

File: source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+工作经历概述：
+* 2011年，云计算售前技术支持，SWG，IBM
+* 2012年，哈尔滨铁路局资产管理项目，包括铁道、桥、隧和涵等，IBM Maximo
+* 2013 - 2014年，IBM BPM Cloud，基于云原生打造下一代业务流程云，方案为 CloudFoundry(Docker 流行之前的最主要云原生方案)
+* 2015年，BJ NodeJS Club 开源社区建设，组织并运营开发者交流活动
+* 2016年，儿童英语聊天机器人，使用 TensorFlow 训练基于深度学习的语言模型
+* 2017年，海通证券、广发证券、安永的机器人客服
+* 2018年至今，开源客服系统，春松客服；Chatopera 云服务，低代码和无服务器上线聊天机器人
+
+
```
Comment (tisonkun) [inline source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:113]:
```suggestion
* 2011 年，云计算售前技术支持，SWG，IBM
* 2012 年，哈尔滨铁路局资产管理项目，包括铁道、桥、隧和涵等，IBM Maximo
* 2013-2014 年，IBM BPM Cloud，基于云原生打造下一代业务流程云，方案为 CloudFoundry（Docker 流行之前的最主要云原生方案）
* 2015 年，BJ NodeJS Club 开源社区建设，组织并运营开发者交流活动
* 2016 年，儿童英语聊天机器人，使用 TensorFlow 训练基于深度学习的语言模型
* 2017 年，海通证券、广发证券、安永的机器人客服
* 2018 年至今，开源客服系统，春松客服；Chatopera 云服务，低代码和无服务器上线聊天机器人

```

File: source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+## 释放中小企业需求
+
+一个模式实现为上千万企业交付企业软件，比如智能客服系统，还没有出现，在中国，2021 年腾讯启点客服被评选为最大的智能客服系统提供商，其公布的企业客户数是 100 万，其采用的模式是 SaaS。那么有那么多企业愿意购置开源加云原生的企业软件吗？在欧美，2021 年 Odoo 在 CRM 和 ERP 领域，通过开源加云原生模式，做到了 700 万的企业客户，Odoo 从 2005 年开始做开源软件，2011 年将重点转向合作伙伴网络，2013 年上线 SaaS，2018 年上线开源加云原生方案，2019年和 2020 年开始高速增长，2020 年有 450万用户，2021 年达到 700 万用户。这个例子，是一个侧面验证，那么我们该如何理解这个例子？这个例子可以说明，孙正义提出的时间机器理论，同样适用于先进的企业和落后的企业之间的数字化进程。
+
+孙正义的时间机器理论说，充分利用不同国家和行业发展的不平衡，在发达市场获取经验之后，再去开发相对落后的市场，就仿佛坐上了时空机，能穿越过去和未来。就拿支付软件来讲，美国 Paypal 诞生于 1998 年，随后，中国支付宝 2003 年出现，再者印度 Paytm 于 2010 年左右首次出现。在移动通讯方面，美国 WhatsApp 2009年，韩国 TALK 2010年，中国微信 2011 年，日本 LINE 则是 2011年。而电子商务呢，美国 Amazon 1995 年，日本乐天 1997 年，中国阿里巴巴 1999 年，印度 Flipkart 2007 年。所以，大家认可时间机器理论，事件发展有客观规律。
+
+将时间机器运用到企业数字化中，就是==先进企业的数字化经验，同样会被落后的企业走一下==，如果落后的企业想要经营下去。 那么，先进的企业的数字化经验是什么呢？最主要的就是使用**定制化的软件**，根据自身业务的特点定制软件。先进的企业会不断重塑自身，通过大大小小的创新。中小企业也是明白这个道理，中小企业也想创新，要定制软件。比如在客服系统上，从国内招投标平台，可以看到政府机构、央企、国企、500强企业们在智能客服系统的招投标项目非常多，有新建有改造有维护，种类繁多，项目繁多。这些同样会发生在中小型企业，它们没有发生是因为过去的供给方的模式，没有精细化分工，没有飞轮效应，就不能释放中小企业的需求。
+
+我们还可以想到，在没有发明汽车以前，人们坐马车出行，能出远门的人很少，多数是富有的人才旅游，而汽车出现后，大家都可以出远门旅游了，这时候旅游的人就多了。
+
```
Comment (tisonkun) [inline source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:97]:
```suggestion
```

**Review by tisonkun — COMMENTED** [review]
Generally looks good. Comments inline.

**Comment by hailiang-wang** [comment]
With the above changes in the morning, my work is finished. 
Please help review. @tisonkun

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks for your contribution! Merging...

**Comment by hailiang-wang** [comment]
My pleasure.

File: source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
+---
+title: 开源加云原生：企业软件的大规模进化之路
+date: 2022-03-23
+author: hailiang-wang
+categories: hailiang-wang
```
Comment (tisonkun) [inline source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:5]: @hailiang-wang would you mind I change the author field to "王海良"? It's a personal preference and it's OK to decline.

Comment (hailiang-wang) [reply source/_posts/oss-plus-cloudnative-large-dist-enterprise-software.md:5]: Sure thing, thanks.

**Pull request merged**